### PR TITLE
Add documentation for Prometheus metrics in Training Operator

### PR DESCRIPTION
**Pull request by sophie0730**
Follow up from [https://github.com/kubeflow/training-operator/issues/2254](https://github.com/kubeflow/training-operator/issues/2254)

## Description
This PR adds a new section to the documentation explaining how to monitor Kubeflow training jobs using Prometheus metrics exposed by the Training Operator.

## Changes
- Rewrote and added a new page titled "Prometheus Monitoring" under the Training Operator documentation.
- Explained how to access the Prometheus metrics for the Training Operator.
- Provided a detailed list of relevant metrics, including description and label information.

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

File: content/en/docs/components/training/prometheus.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+weight = 70
++++
+
+This guide explains how to monitor Kubeflow training jobs using Prometheus metrics. The Training Operator exposes these metrics, providing essential insights into the status of distributed machine learning workloads.
+
+### Prometheus Metrics for Training Operator
+The Training Operator includes a built-in `/metrics` endpoint exposes Prometheus metrics. This feature is enabled by default and requires no additional configuration for basic use.
+
+#### Accessing the Metrics
+By default, the metrics are exposed on port 8080. The method to access these metrics may vary depending on your Kubernetes setup and environment.
```
Comment (andreyvelich) [inline content/en/docs/components/training/prometheus.md:13]: Please explain how to change the default port for metrics exporting: https://github.com/kubeflow/training-operator/blob/master/cmd/training-operator.v1/main.go#L83

File: content/en/docs/components/training/prometheus.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
++++
+title = "Prometheus Monitoring"
```
Comment (andreyvelich) [inline content/en/docs/components/training/prometheus.md:2]: Please can you add this guide under user guides ?

File: content/en/docs/components/training/prometheus.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+| `training_operator_jobs_deleted_total`   |  Total number of jobs deleted       | `namespace`, `framework`                 |
+| `training_operator_jobs_successful_total` |  Total number of successful jobs   |  `namespace`, `framework`                 |
+| `training_operator_jobs_failed_total`    |  Total number of failed jobs       |  `namespace`, `framework` |
+| `training_operator_jobs_restarted_total` |  Total number of restarted jobs   |  `namespace`, `framework`|
+
+Labels information can be interpreted as follow:
+| Label name                          |  Description                     | 
+|------------------------------------|---------|--------------------------|
+| `namespace`   | The Kubernetes namespace where the job is running        |
+| `framework` | The machine learning framework used(e.g. TensorFlow,PyTorch)     |
```
Comment (andreyvelich) [inline content/en/docs/components/training/prometheus.md:41]:
```suggestion
| `framework` | The machine learning framework used (e.g. TensorFlow,PyTorch)     | 
```

File: content/en/docs/components/training/prometheus.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Labels information can be interpreted as follow:
+| Label name                          |  Description                     | 
+|------------------------------------|---------|--------------------------|
+| `namespace`   | The Kubernetes namespace where the job is running        |
+| `framework` | The machine learning framework used(e.g. TensorFlow,PyTorch)     | 
+
+
+Important: Metrics are only generated in response to specific events. For example, job creation metrics will only appear after a job has been created. If a metric is not visible, it may be because the corresponding event has not occurred yet.
+
+These metrics help you understand how your training jobs are doing. You can use this information to fix problems and make your jobs run better.
```
Comment (andreyvelich) [inline content/en/docs/components/training/prometheus.md:46]: I think, you can move this statement to the beginning of this doc.

File: content/en/docs/components/training/prometheus.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+| `training_operator_jobs_restarted_total` |  Total number of restarted jobs   |  `namespace`, `framework`|
+
+Labels information can be interpreted as follow:
+| Label name                          |  Description                     | 
+|------------------------------------|---------|--------------------------|
+| `namespace`   | The Kubernetes namespace where the job is running        |
+| `framework` | The machine learning framework used(e.g. TensorFlow,PyTorch)     | 
+
+
+Important: Metrics are only generated in response to specific events. For example, job creation metrics will only appear after a job has been created. If a metric is not visible, it may be because the corresponding event has not occurred yet.
```
Comment (andreyvelich) [inline content/en/docs/components/training/prometheus.md:44]: If you want you can use the Note template, like here: https://www.kubeflow.org/docs/about/membership/#:~:text=Members-,Note,-Detailed%20documentation%20for

**Review by andreyvelich — COMMENTED** [review]
Thank you for this @sophie0730!
@kubeflow/wg-training-leads @StefanoFioravanzo @hbelmiro Please take a look.

File: content/en/docs/components/training/prometheus.md (hunk 4 above)
Comment (sophie0730) [reply content/en/docs/components/training/prometheus.md:46]: Thank you for the suggestion! After reviewing, I think this sentence(line 46) seems to duplicate information already mentioned at the beginning of the document. To avoid redundancy, I’ve decided to remove it. Thanks!

File: content/en/docs/components/training/prometheus.md (hunk 5 above)
Comment (sophie0730) [reply content/en/docs/components/training/prometheus.md]: I've incorporated the Note template, thanks for the suggestion!

File: content/en/docs/components/training/prometheus.md (hunk 2 above)
Comment (sophie0730) [reply content/en/docs/components/training/prometheus.md]: Sure. Have moved this guide under user guides section.

File: content/en/docs/components/training/user-guides/prometheus.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+This guide explains how to monitor Kubeflow training jobs using Prometheus metrics. The Training Operator exposes these metrics, providing essential insights into the status of distributed machine learning workloads.
+
+{{< note >}}
+Metrics are only generated in response to specific events. For example, job creation metrics will only appear after a job has been created. If a metric is not visible, it may be because the corresponding event has not occurred yet.
+{{< /note >}}
+
+## Prometheus Metrics for Training Operator
+The Training Operator includes a built-in `/metrics` endpoint exposes Prometheus metrics. This feature is enabled by default and requires no additional configuration for basic use.
+
+### Configuring Metrics Port
```
Comment (sophie0730) [inline content/en/docs/components/training/user-guides/prometheus.md:16]: @andreyvelich Thank you for the suggestion. I've added the section about how to configure the default port for metrics exporting. Please let me know if there are any comments.

File: content/en/docs/components/training/user-guides/prometheus.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        image: kubeflow/training-operator
+        name: training-operator
+        ports:
+        - containerPort: 8080
+        - containerPort: 9443
+            name: webhook-server
+            protocol: TCP
+        args:
+        - "--metrics-bind-address=:8082" # Metrics port changed to 8082
+```
```
Comment (sophie0730) [inline content/en/docs/components/training/user-guides/prometheus.md:35]: @andreyvelich Thank you for the suggestion. I've added a section on how to configure the default port for metrics exporting. Please let me know if you have further comments.

File: content/en/docs/components/training/prometheus.md (hunk 1 above)
Comment (sophie0730) [reply content/en/docs/components/training/prometheus.md]: I've already added the information on changing the default port for metrics exporting. Thanks.

**Review by hbelmiro — COMMENTED** [review]
/lgtm

File: content/en/docs/components/training/user-guides/prometheus.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+{{< note >}}
+Metrics are only generated in response to specific events. For example, job creation metrics will only appear after a job has been created. If a metric is not visible, it may be because the corresponding event has not occurred yet.
+{{< /note >}}
+
+## Prometheus Metrics for Training Operator
+The Training Operator includes a built-in `/metrics` endpoint exposes Prometheus metrics. This feature is enabled by default and requires no additional configuration for basic use.
+
+### Configuring Metrics Port
+By default, metrics are exposed on port 8080. If you want to change the default port for metrics exporting, simply add the `metrics-bind-address` argument.
```
Comment (terrytangyuan) [inline content/en/docs/components/training/user-guides/prometheus.md:17]: The default is only available if it's enabled though, right?

**Review by terrytangyuan — COMMENTED** [review]
This is great to have!

File: content/en/docs/components/training/user-guides/prometheus.md (hunk 8 above)
Comment (tenzen-y) [inline content/en/docs/components/training/user-guides/prometheus.md:17]: Could we mention that the metrics can be scraped from any IP address by default?
And then we can mention that limiting the IP address by "x.x.x.x:xxxx".

File: content/en/docs/components/training/user-guides/prometheus.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+| `training_operator_jobs_restarted_total` |  Total number of restarted jobs   |  `namespace`, `framework`|
+
+Labels information can be interpreted as follows:
+| Label name                          |  Description                     | 
+|------------------------------------|---------|--------------------------|
+| `namespace`   | The Kubernetes namespace where the job is running        |
+| `framework` | The machine learning framework used (e.g. TensorFlow,PyTorch)     | 
+
+
+
```
Comment (tenzen-y) [inline content/en/docs/components/training/user-guides/prometheus.md:66]:
```suggestion
```

**Review by tenzen-y — COMMENTED** [review]
Thanks for doing this.

File: content/en/docs/components/training/user-guides/prometheus.md (hunk 8 above)
Comment (sophie0730) [reply content/en/docs/components/training/user-guides/prometheus.md]: I have updated this section in my latest commit. Please let me know if you have any further questions! Thank you for your suggestion.

Comment (sophie0730) [reply content/en/docs/components/training/user-guides/prometheus.md]: I think you are right. The metrics can only be scraped from the /metrics endpoint if this feature is enabled.

**Review by andreyvelich — COMMENTED** [review]
Thanks for the update @sophie0730!
Please can you submit PR to remove that doc from Training Operator: https://github.com/kubeflow/training-operator/tree/master/docs/monitoring
/lgtm
/assign @tenzen-y @terrytangyuan

**Review by tenzen-y — COMMENTED** [review]
Thank you for this contribution!
This is really helpful for both cluster admins and batch users!
/lgtm
/approve

**Comment by sophie0730** [comment]
Thanks for the review @andreyvelich and @tenzen-y !

I have created a PR for removing the doc from the Training Operator repo: https://github.com/kubeflow/training-operator/pull/2301
